src/htmx.js

```diff
@@ -1141,31 +1141,51 @@ var htmx = (function() {
    */
   function querySelectorAllExt(elt, selector, global) {
     elt = resolveTarget(elt)
-    if (selector.indexOf('closest ') === 0) {
-      return [closest(asElement(elt), normalizeSelector(selector.substr(8)))]
-    } else if (selector.indexOf('find ') === 0) {
-      return [find(asParentNode(elt), normalizeSelector(selector.substr(5)))]
-    } else if (selector === 'next') {
-      return [asElement(elt).nextElementSibling]
-    } else if (selector.indexOf('next ') === 0) {
-      return [scanForwardQuery(elt, normalizeSelector(selector.substr(5)), !!global)]
-    } else if (selector === 'previous') {
-      return [asElement(elt).previousElementSibling]
-    } else if (selector.indexOf('previous ') === 0) {
-      return [scanBackwardsQuery(elt, normalizeSelector(selector.substr(9)), !!global)]
-    } else if (selector === 'document') {
-      return [document]
-    } else if (selector === 'window') {
-      return [window]
-    } else if (selector === 'body') {
-      return [document.body]
-    } else if (selector === 'root') {
-      return [getRootNode(elt, !!global)]
-    } else if (selector.indexOf('global ') === 0) {
-      return querySelectorAllExt(elt, selector.slice(7), true)
-    } else {
-      return toArray(asParentNode(getRootNode(elt, !!global)).querySelectorAll(normalizeSelector(selector)))
+
+    const parts = selector.split(',')
+    const result = []
+    const unprocessedParts = []
+    for (let i = 0; i < parts.length; i++) {
+      let selector = parts[i].trimStart()
+
+      if (selector.indexOf('global ') === 0) {
+        global = true
```

Comment:
overriding global here will impact all later iterations making them global as well.  not likely in practice but could be fixed just in case anyway.

Reply:
Actually maybe don't need to fix this as supporting full global and non global flexibility is probably not a real useful feature and it will take a lot of work to solve this properly. Or you would have to remove your unprocessedParts method and rename and then wrap the original function unchanged with an outer for loop.  

It would still support having global at the start like it works now so won't break any existing use cases even if we don't handle every edge case.

Reply:
> overriding global here will impact all later iterations

Yeah I wanted to preserve the old behavior where `global` at the start of a selector would apply to all following selectors.

I wondered a lot about this, and I'm still unsure about what we should strive for functionnally:
- do we keep `global` applying to all selectors if put at the start, but only to the selector it's put in front of if it's not in first position ? 
- do we keep `global` applying to all selectors if put at the start, and to all subsequent selectors no matter its position ?

It's a bummer that we have to maintain this "`global` first = apply to all" for backwards compatibility but we can't help that. Still unsure on the way to go about this. When I wrote this code I felt that the second option was the most logical (or the less nonsensical), now I'm not sure anymore 😆

Reply:
All your changes are looking good to me.  I think you can leave the global handling as you have it now.  It should be backwards compatible. If someone does need to do local web component scoped selector plus global scoped at the same time you can still do `find input, global button` and it will select a local web component scoped input plus all document scoped buttons for example.  the only issue is if someone does `input, global button` it will not work as they may expect as input moves to global scope because non extended selectors will go in unprocessedParts and now be global.

If we wanted to support global not applying to earlier non extended selectors like this you could remove the unprocessedParts fallback like this:
```javascript
      } else if (selector === 'root') {
        item = getRootNode(elt, global)
      } else {
        const rootNode = asParentNode(getRootNode(elt, global))
        result.push(...toArray(rootNode.querySelectorAll(selector)))
      }

      if (item) {
        result.push(item)
      }
    }

    return result
```
Or make it more complex with unprocessedGlobalParts[] as well.

Reply:
> it will not work as they may expect as input moves to global scope because non extended selectors will go in unprocessedParts and now be global.

Ooooh right, totally overlooked that. Indeed, that's not ideal, I'll fix this to at least work in a predictable manner for users

Reply:
Alright should be fine now @MichaelWest22, I also added a test to ensure that `global` doesn't apply to previous selectors.
I didn't go for the "push at every iteration" approach, as it would break nested standard selectors (see the added `'hx-include processes nested standard selectors correctly'` test).
As the initial selectors string is split on commas, this would break the following standard selector in 2 unusable halves if not joined back in a single string to pass to `querySelectorAll` : `:is([name='i1'], [name='i2'])`

Reply:
Yeah that looks to work and cover all the cases well now!  You had to add some duplication and complexity which is why you need such big inline comments.  I Just tried rewriting it by turning the for parts[i] loop using a while loop with shift/unshift and i got this: https://github.com/MichaelWest22/htmx/commit/1b4a7949f9b8584ab08695b22366f066d2cc9c21

It follows the original global recursive pattern unchanged but seems to handle everything well.  The only functional change is that it adds global results first before standard selector results but i don't think result return order is a breaking change.  Anyway you can use this and add any required inline comments or just go with what you have already that works fine.

Reply:
Oh, very smart @MichaelWest22 indeed.

> The only functional change is that it adds global results first before standard selector results but i don't think result return order is a breaking change

I don't it is either, the only case I can imagine is if you want to preserve values order for multiple inputs that share the same `name`, but for it to be an issue, that would mean you have some of them in the shadow dom and some other outside... Sounds like a bit too-convoluted use-case.
Anyway, if that's an issue for someone, it can always be fixed later on (after all, we're not breaking anything, given that `global` not in first position wasn't working at all formerly!)

The other advantage of your suggestion here is that it keeps the diff minimal compared to the original code, which will greatly improve the chances of this PR to get reviewed & merged😁 

Thanks a lot for your help on this!